src/main/js/Delete/FileDeleteButton.tsx

```diff
@@ -93,8 +93,24 @@ class FileDeleteButton extends React.Component<Props, State> {
           const filePath = location.pathname
             .substr(0, location.pathname.length - file.name.length - 1)
             .split("/sources/" + revision)[1];
+
+          const filePathParts = filePath.split("/");
+          const checkFolderBaseUrl =
+            (newCommit._links.self as Link).href.split("/changesets/")[0] +
+            `/sources/${encodeURIComponent(newRevision)}`;
+          let exists = false;
+          while (!exists) {
+            try {
+              await apiClient.get(checkFolderBaseUrl + filePathParts.join("/"));
+              exists = true;
+            } catch (err) {
+              filePathParts.pop();
+            }
+          }
```

Comment:
It works like a charm. I would extract this part into a new function like `resolveRedirectPath` to improve the readability.